### PR TITLE
Cholesky decomposition NotImplementedError and test

### PR DESCRIPTION
**Pull request by LuiGiovanni**
This is in reference to #852 

The first PR is about adding the NotImplementedError and the respective test function, forgive all these PR and branches it is the first time I have worked on this.

### CODE REVIEW

File: tensornetwork/backends/abstract_backend.py [hunk 1]
```diff
@@ -1022,3 +1022,11 @@ def item(self, tensor) -> Union[float, int, complex]:
       The value in tensor.
     """
     raise NotImplementedError("Backend {self.name} has not implemented item")
+
+  def cholesky_decomposition(self, 
+                             tensor: Tensor,
+                             pivot_axis: int = -1,
```
Comment (alewis) [inline tensornetwork/backends/abstract_backend.py:1028]: Let's just call it "cholesky" like in numpy

**Review by mganahl — APPROVED** [review]
looks good mod @alewis comments

File: tensornetwork/backends/abstract_backend.py (hunk 1 above)
Comment (LuiGiovanni) [reply tensornetwork/backends/abstract_backend.py:1028]: Very well, I will implement it right now!

**Comment by LuiGiovanni** [comment]
@alewis The function name has been changed.

**Comment by LuiGiovanni** [comment]
@alewis Thank you for everything, I will start adding the Cholesky function to the decomposition files, should I open a new issue for this or continue referencing this issue?

**Pull request merged**